File: wagerproof-ios-native/Wagerproof/Features/Props/Components/NFLPropSignalViews.swift
import SwiftUI
import WagerproofModels
import WagerproofDesign

// MARK: - Prop signal UI (P1–P18 rule flags)

/// Compact prop-signal row shown beneath an NFL prop feed card when the
/// displayed market fired one or more P-flags.
struct NFLPropSignalFeedStrip: View {
    let flags: [String]
    /// Optional backtest records keyed by flag (`P14` → `12-8 · 60%`) from the
    /// pages payload — shown when present so the strip matches web chips.
    var recordsByKey: [String: String] = [:]

    private var signals: [NFLPropSignalDefinition] {
        NFLPropSignalDefinitions.resolve(flags)
    }

    private var actionable: [NFLPropSignalDefinition] {
        signals.filter { !$0.isAntiSignal }
    }

    private var anti: [NFLPropSignalDefinition] {
        signals.filter(\.isAntiSignal)
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            header
            if !actionable.isEmpty {
                signalGroup(title: "Supports this prop", signals: actionable, muted: false)
            }
            if !anti.isEmpty {
                signalGroup(title: "Avoid this prop", signals: anti, muted: true)
            }
        }
        .padding(.horizontal, 10)
        .padding(.vertical, 9)
        .background(Color.appSurfaceElevated.opacity(0.55), in: RoundedRectangle(cornerRadius: 14, style: .continuous))
        .overlay(
            RoundedRectangle(cornerRadius: 14, style: .continuous)
                .stroke(Color.appBorder.opacity(0.45), lineWidth: 0.6)
        )
    }

    private var header: some View {
        HStack(spacing: 6) {
            Image(systemName: "bolt.fill")
                .font(.system(size: 11, weight: .black))
                .foregroundStyle(Color(hex: 0xF97316))
            Text(signals.count == 1 ? "1 Prop Signal" : "\(signals.count) Prop Signals")
                .font(.system(size: 11, weight: .black))
                .foregroundStyle(Color(hex: 0xF97316))
            Spacer(minLength: 0)
        }
    }

    @ViewBuilder
    private func signalGroup(title: String, signals: [NFLPropSignalDefinition], muted: Bool) -> some View {
        VStack(alignment: .leading, spacing: 6) {
            Text(title)
                .font(.system(size: 9, weight: .black))
                .foregroundStyle(muted ? Color.appAccentAmber : Color.appTextMuted)
            VStack(spacing: 6) {
                ForEach(signals) { signal in
                    NFLPropSignalCompactRow(
                        signal: signal,
                        muted: muted,
                        record: recordsByKey[signal.id]
                    )
                }
            }
        }
    }
}

private struct NFLPropSignalCompactRow: View {
    let signal: NFLPropSignalDefinition
    let muted: Bool
    var record: String? = nil

    private var tint: Color { muted ? Color.appAccentAmber : Color.appAccentBlue }

    var body: some View {
        HStack(spacing: 8) {
            Image(systemName: muted ? "exclamationmark.triangle.fill" : "info.circle.fill")
                .font(.system(size: 11, weight: .black))
                .foregroundStyle(tint)
            VStack(alignment: .leading, spacing: 1) {
                Text(signal.displayName)
                    .font(.system(size: 11, weight: .black))
                    .foregroundStyle(tint)
                    .lineLimit(1)
                    .minimumScaleFactor(0.8)
                HStack(spacing: 6) {
                    Text(signal.betDirection)
                        .font(.system(size: 9, weight: .heavy))
                        .foregroundStyle(tint.opacity(0.75))
                    if let record, !record.isEmpty {
                        Text(record)
                            .font(.system(size: 9, weight: .semibold, design: .monospaced))
                            .foregroundStyle(tint.opacity(0.7))
                            .lineLimit(1)
                    }
                }
            }
            Spacer(minLength: 0)
        }
        .padding(.horizontal, 10)
        .padding(.vertical, 7)
        .background(tint.opacity(muted ? 0.12 : 0.16), in: RoundedRectangle(cornerRadius: 10, style: .continuous))
        .overlay(
            RoundedRectangle(cornerRadius: 10, style: .continuous)
                .stroke(tint.opacity(muted ? 0.45 : 0.38), lineWidth: 0.8)
        )
    }
}

struct NFLPropSignalGroup: View {
    let flags: [String]
    var recordsByKey: [String: String] = [:]
    var onSelect: (NFLPropSignalDefinition) -> Void = { _ in }

    private var signals: [NFLPropSignalDefinition] {
        NFLPropSignalDefinitions.resolve(flags)
    }

    var body: some View {
        if signals.isEmpty {
            EmptyView()
        } else {
            let actionable = signals.filter { !$0.isAntiSignal }
            let anti = signals.filter(\.isAntiSignal)
            VStack(alignment: .leading, spacing: 9) {
                if !actionable.isEmpty {
                    detailGroup(title: "Supports this prop", signals: actionable, muted: false)
                }
                if !anti.isEmpty {
                    detailGroup(title: "Avoid this prop", signals: anti, muted: true)
                }
            }
        }
    }

    @ViewBuilder
    private func detailGroup(title: String, signals: [NFLPropSignalDefinition], muted: Bool) -> some View {
        VStack(alignment: .leading, spacing: 7) {
            Text(title)
                .font(.system(size: 9, weight: .black))
                .foregroundStyle(muted ? Color.appAccentAmber : Color.appTextMuted)
            // Full-width stacked rows — the old adaptive grid squeezed chips to
            // ~118pt and truncated titles like "Featured WR".
            VStack(spacing: 8) {
                ForEach(signals) { signal in
                    NFLPropSignalButton(
                        signal: signal,
                        muted: muted,
                        record: recordsByKey[signal.id],
                        onSelect: onSelect
                    )
                }
            }
        }
    }
}

/// Emerald chip row matching web `PropBreakdownPage` — label + optional record
/// when we have the pages payload but may not yet have catalog definitions.
struct NFLPropSignalChipRow: View {
    let signals: [NFLPropPageSignal]
    var onSelect: (NFLPropPageSignal) -> Void = { _ in }

    var body: some View {
        if signals.isEmpty {
            EmptyView()
        } else {
            FlowLayout(spacing: 6) {
                ForEach(signals) { sig in
                    Button { onSelect(sig) } label: {
                        HStack(spacing: 5) {
                            Text("⚡ \(sig.label)")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundStyle(Color(hex: 0x059669))
                            if let record = sig.record, !record.isEmpty {
                                Text(record)
                                    .font(.system(size: 10, weight: .semibold, design: .monospaced))
                                    .foregroundStyle(Color(hex: 0x047857).opacity(0.85))
                            }
                        }
                        .padding(.horizontal, 10)
                        .padding(.vertical, 6)
                        .background(Color(hex: 0x10B981).opacity(0.12), in: Capsule())
                        .overlay(Capsule().stroke(Color(hex: 0x10B981).opacity(0.4), lineWidth: 1))
                    }
                    .buttonStyle(.plain)
                }
            }
        }
    }
}

private struct NFLPropSignalButton: View {
    let signal: NFLPropSignalDefinition
    let muted: Bool
    var record: String? = nil
    let onSelect: (NFLPropSignalDefinition) -> Void

    private var color: Color { muted ? Color.appAccentAmber : Color.appAccentBlue }

    var body: some View {
        Button { onSelect(signal) } label: {
            HStack(spacing: 10) {
                Image(systemName: muted ? "exclamationmark.triangle.fill" : "info.circle.fill")
                    .font(.system(size: 13, weight: .black))
                    .foregroundStyle(color)
                VStack(alignment: .leading, spacing: 3) {
                    Text(signal.displayName)
                        .font(.system(size: 14, weight: .bold))
                        .foregroundStyle(color)
                        .lineLimit(2)
                        .multilineTextAlignment(.leading)
                        .fixedSize(horizontal: false, vertical: true)
                    HStack(spacing: 6) {
                        Text(signal.betDirection)
                            .font(.system(size: 11, weight: .heavy))
                            .foregroundStyle(color.opacity(0.78))
                        if let record, !record.isEmpty {
                            Text("·")
                                .font(.system(size: 11, weight: .bold))
                                .foregroundStyle(color.opacity(0.45))
                            Text(record)
                                .font(.system(size: 11, weight: .semibold, design: .monospaced))
                                .foregroundStyle(color.opacity(0.72))
                                .lineLimit(1)
                        }
                    }
                }
                Spacer(minLength: 8)
                Image(systemName: "chevron.right")
                    .font(.system(size: 11, weight: .bold))
                    .foregroundStyle(color.opacity(0.85))
            }
            .padding(.horizontal, 12)
            .padding(.vertical, 11)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(color.opacity(muted ? 0.10 : 0.14), in: RoundedRectangle(cornerRadius: 12, style: .continuous))
            .overlay(
                RoundedRectangle(cornerRadius: 12, style: .continuous)
                    .stroke(color.opacity(muted ? 0.45 : 0.38), lineWidth: 1)
            )
            .contentShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
        }
        .buttonStyle(.plain)
    }
}

struct NFLPropSignalDetailSheet: View {
    let signal: NFLPropSignalDefinition
    let seasonRecord: SignalPerformance?
    @Environment(\.dismiss) private var dismiss

    var body: some View {
        NavigationStack {
            ScrollView {
                VStack(alignment: .leading, spacing: 16) {
                    Text(signal.displayName)
                        .font(.system(size: 22, weight: .black))
                        .foregroundStyle(Color.appTextPrimary)
                    if !signal.oneLiner.isEmpty {
                        Text(signal.oneLiner)
                            .font(.system(size: 14, weight: .medium))
                            .foregroundStyle(Color.appTextSecondary)
                    }
                    signalBlock("Definition", signal.definition)
                    signalBlock("Why It Works", signal.whyItWorks)
                    signalBlock("Bet Direction", signal.betDirection)
                    SignalPerformanceStatsSection(
                        backtestHit: signal.typicalHit,
                        seasonDisplay: SignalSeasonRecordDisplay(performance: seasonRecord)
                    )
                    if signal.isAntiSignal {
                        Text("This is an anti-signal — the backtest says to avoid betting this market when it fires.")
                            .font(.system(size: 13, weight: .medium))
                            .foregroundStyle(Color.appAccentAmber)
                    }
                }
                .padding(20)
            }
            .background(Color.appSurface)
            .navigationTitle("Prop Signal")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .topBarTrailing) {
                    Button("Done") { dismiss() }
                        .tint(Color.appPrimary)
                }
            }
        }
        .presentationDetents([.medium, .large])
        .presentationDragIndicator(.visible)
    }

    @ViewBuilder
    private func signalBlock(_ title: String, _ body: String) -> some View {
        if !body.isEmpty {
            VStack(alignment: .leading, spacing: 6) {
                Text(title.uppercased())
                    .font(.system(size: 10, weight: .black))
                    .tracking(0.6)
                    .foregroundStyle(Color.appTextMuted)
                Text(body)
                    .font(.system(size: 14))
                    .foregroundStyle(Color.appTextPrimary)
                    .fixedSize(horizontal: false, vertical: true)
            }
        }
    }
}

/// Simple wrapping layout for emerald signal chips.
private struct FlowLayout: Layout {
    var spacing: CGFloat = 6

    func sizeThatFits(proposal: ProposedViewSize, subviews: Subviews, cache: inout ()) -> CGSize {
        let maxWidth = proposal.width ?? .infinity
        var x: CGFloat = 0
        var y: CGFloat = 0
        var rowHeight: CGFloat = 0
        var height: CGFloat = 0
        for sub in subviews {
            let size = sub.sizeThatFits(.unspecified)
            if x + size.width > maxWidth, x > 0 {
                x = 0
                y += rowHeight + spacing
                rowHeight = 0
            }
            rowHeight = max(rowHeight, size.height)
            x += size.width + spacing
            height = y + rowHeight
        }
        return CGSize(width: maxWidth.isFinite ? maxWidth : x, height: height)
    }

    func placeSubviews(in bounds: CGRect, proposal: ProposedViewSize, subviews: Subviews, cache: inout ()) {
        var x = bounds.minX
        var y = bounds.minY
        var rowHeight: CGFloat = 0
        for sub in subviews {
            let size = sub.sizeThatFits(.unspecified)
            if x + size.width > bounds.maxX, x > bounds.minX {
                x = bounds.minX
                y += rowHeight + spacing
                rowHeight = 0
            }
            sub.place(at: CGPoint(x: x, y: y), proposal: ProposedViewSize(size))
            x += size.width + spacing
            rowHeight = max(rowHeight, size.height)
        }
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/props/components/NflPropSignalViews.kt
package com.wagerproof.app.features.props.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.Text
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.core.design.icons.AppIcon
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.models.NFLPropSignalDefinition
import com.wagerproof.core.models.NFLPropSignalDefinitions
import com.wagerproof.core.models.SignalPerformance
import com.wagerproof.core.models.SignalSeasonRecordDisplay

private val SignalOrange = Color(0xFFF97316)

/** Compact prop-signal rows shown beneath an NFL feed card when the market fired flags. */
@Composable
fun NFLPropSignalFeedStrip(flags: List<String>) {
    val signals = NFLPropSignalDefinitions.resolve(flags)
    if (signals.isEmpty()) return
    val actionable = signals.filter { !it.isAntiSignal }
    val anti = signals.filter { it.isAntiSignal }
    val shape = RoundedCornerShape(14.dp)

    Column(
        Modifier
            .fillMaxWidth()
            .clip(shape)
            .background(AppColors.appSurfaceElevated.copy(alpha = 0.55f))
            .border(0.6.dp, AppColors.appBorder.copy(alpha = 0.45f), shape)
            .padding(horizontal = 10.dp, vertical = 9.dp),
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(6.dp)) {
            Icon(AppIcon.BOLT_FILL.imageVector, null, tint = SignalOrange, modifier = Modifier.size(11.dp))
            Text(
                if (signals.size == 1) "1 Prop Signal" else "${signals.size} Prop Signals",
                color = SignalOrange, fontSize = 11.sp, fontWeight = FontWeight.Black,
            )
        }
        if (actionable.isNotEmpty()) SignalGroup("Supports this prop", actionable, muted = false)
        if (anti.isNotEmpty()) SignalGroup("Avoid this prop", anti, muted = true)
    }
}

@Composable
private fun SignalGroup(title: String, signals: List<NFLPropSignalDefinition>, muted: Boolean) {
    Column(verticalArrangement = Arrangement.spacedBy(6.dp)) {
        Text(title, color = if (muted) AppColors.appAccentAmber else AppColors.appTextMuted, fontSize = 9.sp, fontWeight = FontWeight.Black)
        signals.forEach { SignalCompactRow(it, muted) }
    }
}

@Composable
private fun SignalCompactRow(signal: NFLPropSignalDefinition, muted: Boolean) {
    val tint = if (muted) AppColors.appAccentAmber else AppColors.appAccentBlue
    val shape = RoundedCornerShape(10.dp)
    Row(
        Modifier
            .fillMaxWidth()
            .clip(shape)
            .background(tint.copy(alpha = if (muted) 0.12f else 0.16f))
            .border(0.8.dp, tint.copy(alpha = if (muted) 0.45f else 0.38f), shape)
            .padding(horizontal = 10.dp, vertical = 7.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Icon(
            (if (muted) AppIcon.fromSystemName("exclamationmark.triangle.fill") else AppIcon.fromSystemName("info.circle.fill"))?.imageVector
                ?: AppIcon.INFO_CIRCLE_FILL.imageVector,
            null, tint = tint, modifier = Modifier.size(11.dp),
        )
        Column {
            Text(signal.displayName, color = tint, fontSize = 11.sp, fontWeight = FontWeight.Black, maxLines = 1)
            Text(signal.betDirection, color = tint.copy(alpha = 0.75f), fontSize = 9.sp, fontWeight = FontWeight.ExtraBold)
        }
    }
}

/** Detail-page variant: an adaptive grid of tappable signal buttons. */
@Composable
fun NFLPropSignalGroup(
    flags: List<String>,
    onSelect: (NFLPropSignalDefinition) -> Unit,
) {
    val signals = NFLPropSignalDefinitions.resolve(flags)
    if (signals.isEmpty()) return
    val actionable = signals.filter { !it.isAntiSignal }
    val anti = signals.filter { it.isAntiSignal }
    Column(verticalArrangement = Arrangement.spacedBy(9.dp)) {
        if (actionable.isNotEmpty()) SignalButtonGroup("Supports this prop", actionable, muted = false, onSelect)
        if (anti.isNotEmpty()) SignalButtonGroup("Avoid this prop", anti, muted = true, onSelect)
    }
}

@Composable
private fun SignalButtonGroup(
    title: String,
    signals: List<NFLPropSignalDefinition>,
    muted: Boolean,
    onSelect: (NFLPropSignalDefinition) -> Unit,
) {
    val color = if (muted) AppColors.appAccentAmber else AppColors.appAccentBlue
    Column(verticalArrangement = Arrangement.spacedBy(7.dp)) {
        Text(title, color = if (muted) AppColors.appAccentAmber else AppColors.appTextMuted, fontSize = 9.sp, fontWeight = FontWeight.Black)
        // Full-width stacked rows so titles like "Featured WR" aren't truncated.
        Column(verticalArrangement = Arrangement.spacedBy(8.dp), modifier = Modifier.fillMaxWidth()) {
            signals.forEach { signal ->
                SignalButton(signal, muted, color, onSelect)
            }
        }
    }
}

@Composable
private fun SignalButton(
    signal: NFLPropSignalDefinition,
    muted: Boolean,
    color: Color,
    onSelect: (NFLPropSignalDefinition) -> Unit,
) {
    val shape = RoundedCornerShape(12.dp)
    Row(
        Modifier
            .fillMaxWidth()
            .clip(shape)
            .background(color.copy(alpha = if (muted) 0.10f else 0.14f))
            .border(1.dp, color.copy(alpha = if (muted) 0.45f else 0.38f), shape)
            .clickable { onSelect(signal) }
            .padding(horizontal = 12.dp, vertical = 11.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(10.dp),
    ) {
        Icon(
            (if (muted) AppIcon.fromSystemName("exclamationmark.triangle.fill") else AppIcon.fromSystemName("info.circle.fill"))?.imageVector
                ?: AppIcon.INFO_CIRCLE_FILL.imageVector,
            null, tint = color, modifier = Modifier.size(13.dp),
        )
        Column(Modifier.weight(1f)) {
            Text(
                signal.displayName,
                color = color,
                fontSize = 14.sp,
                fontWeight = FontWeight.Bold,
                maxLines = 2,
            )
            Row(horizontalArrangement = Arrangement.spacedBy(6.dp), verticalAlignment = Alignment.CenterVertically) {
                Text(signal.betDirection, color = color.copy(alpha = 0.78f), fontSize = 11.sp, fontWeight = FontWeight.ExtraBold)
            }
        }
        Icon(
            AppIcon.CHEVRON_RIGHT.imageVector,
            null,
            tint = color.copy(alpha = 0.85f),
            modifier = Modifier.size(14.dp),
        )
    }
}

/**
 * THE prop-signal sheet — renders BOTH the all-time backtest hit
 * (`signal.typicalHit`) AND the season-to-date record ([SignalSeasonRecordDisplay]),
 * kept visually separate. Port of iOS `NFLPropSignalDetailSheet`.
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun NFLPropSignalDetailSheet(
    signal: NFLPropSignalDefinition,
    seasonRecord: SignalPerformance?,
    onDismiss: () -> Unit,
) {
    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
    ModalBottomSheet(
        onDismissRequest = onDismiss,
        sheetState = sheetState,
        containerColor = AppColors.appSurface,
    ) {
        Column(
            Modifier
                .verticalScroll(rememberScrollState())
                .padding(20.dp)
                .padding(bottom = 24.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp),
        ) {
            Text(signal.displayName, color = AppColors.appTextPrimary, fontSize = 22.sp, fontWeight = FontWeight.Black)
            if (signal.oneLiner.isNotEmpty()) {
                Text(signal.oneLiner, color = AppColors.appTextSecondary, fontSize = 14.sp, fontWeight = FontWeight.Medium)
            }
            SignalBlock("Definition", signal.definition)
            SignalBlock("Why It Works", signal.whyItWorks)
            SignalBlock("Bet Direction", signal.betDirection)
            SignalPerformanceStats(backtestHit = signal.typicalHit, seasonDisplay = SignalSeasonRecordDisplay(seasonRecord))
            if (signal.isAntiSignal) {
                Text(
                    "This is an anti-signal — the backtest says to avoid betting this market when it fires.",
                    color = AppColors.appAccentAmber, fontSize = 13.sp, fontWeight = FontWeight.Medium,
                )
            }
        }
    }
}

@Composable
private fun SignalBlock(title: String, body: String) {
    if (body.isEmpty()) return
    Column(verticalArrangement = Arrangement.spacedBy(6.dp)) {
        Text(title.uppercase(), color = AppColors.appTextMuted, fontSize = 10.sp, fontWeight = FontWeight.Black)
        Text(body, color = AppColors.appTextPrimary, fontSize = 14.sp)
    }
}

/**
 * The all-time backtest hit and the season-to-date record, side by side but
 * clearly separated (see memory: keep the two signal records distinct).
 */
@Composable
private fun SignalPerformanceStats(backtestHit: String?, seasonDisplay: SignalSeasonRecordDisplay) {
    val shape = RoundedCornerShape(14.dp)
    Column(
        Modifier
            .fillMaxWidth()
            .clip(shape)
            .background(AppColors.appSurfaceElevated)
            .border(0.5.dp, AppColors.appBorder.copy(alpha = 0.5f), shape)
            .padding(14.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Column(verticalArrangement = Arrangement.spacedBy(3.dp)) {
            Text("ALL-TIME BACKTEST", color = AppColors.appTextMuted, fontSize = 10.sp, fontWeight = FontWeight.Black)
            Text(
                backtestHit ?: "—",
                color = AppColors.appTextPrimary, fontSize = 15.sp, fontWeight = FontWeight.Bold,
            )
        }
        Box(Modifier.fillMaxWidth().height(0.5.dp).background(AppColors.appBorder.copy(alpha = 0.5f)))
        Column(verticalArrangement = Arrangement.spacedBy(3.dp)) {
            Text("THIS SEASON", color = AppColors.appTextMuted, fontSize = 10.sp, fontWeight = FontWeight.Black)
            val toneColor = when (seasonDisplay.tone) {
                SignalSeasonRecordDisplay.Tone.POSITIVE -> AppColors.appWin
                SignalSeasonRecordDisplay.Tone.NEGATIVE -> AppColors.appLoss
                SignalSeasonRecordDisplay.Tone.NEUTRAL -> AppColors.appTextPrimary
                SignalSeasonRecordDisplay.Tone.EMPTY -> AppColors.appTextMuted
            }
            Text(seasonDisplay.detail, color = toneColor, fontSize = 15.sp, fontWeight = FontWeight.Bold)
            if (seasonDisplay.isSmallSample) {
                Text("Small sample — read with caution.", color = AppColors.appTextMuted, fontSize = 11.sp)
            }
        }
    }
}
